apps/team-awesome-web/src/components/team-members/TeamMembers.tsx

```diff
@@ -1,24 +1,32 @@
 import { useContext, useEffect, useState } from "react";
+import { useQuery } from "@apollo/client";
 
-import * as Styled from "./TeamMembers.styled";
+import { Filter } from "../forms/filter";
+import { GET_TEAM_MEMBERS } from "../../graphql/team-members/queries";
+import { Select } from "../forms/select";
+import { StoreContext } from "../store-context";
 import { TeamMember } from "../team-member";
+import * as Styled from "./TeamMembers.styled";
 import * as Types from "../../types";
-import { StoreContext } from "../store-context";
-import { Select } from "../forms/select";
 
 interface TeamMembersProps {
   errorMessage?: string;
   teamMembers: Types.TeamMember[];
 }
 
-export const TeamMembers = ({ ...props }: TeamMembersProps) => {
+export const TeamMembers = ({ errorMessage, ...props }: TeamMembersProps) => {
   const { teamMembers, setTeamMembers } = useContext(StoreContext);
 
-  const [sortedTeamMembers, setSortedTeamMembers] = useState<Types.TeamMember[]>(props.teamMembers);
+  const { loading, data: teamMemberData } = useQuery(GET_TEAM_MEMBERS, {
+    pollInterval: 0,
+    fetchPolicy: "network-only",
+  });
+  if (!loading) {
+    const { allTeamMembers = {} } = teamMemberData;
+    setTeamMembers(allTeamMembers);
+  }
 
-  useEffect(() => {
-    setTeamMembers(props.teamMembers);
-  }, []);
+  const [sortedTeamMembers, setSortedTeamMembers] = useState(teamMembers);
 
   useEffect(() => {
     setSortedTeamMembers(teamMembers);
```

Comment:
Just looking at this piece of code. Can't you leave this useEffect out? Because your initial state of `sortedTeamMembers` is already `teamMembers`.

Reply:
Yup, removing in favor of the 'regular polling' approach!